packages/blocks/src/_common/components/toggle-switch.ts

```diff
@@ -0,0 +1,82 @@
+import { css, html, LitElement } from 'lit';
+import { state } from 'lit/decorators.js';
+
+const styles = css`
+  :host {
+    display: flex;
+  }
+
+  input[type='checkbox'] {
+    height: 0;
+    width: 0;
+    visibility: hidden;
+    margin: 0;
+  }
+
+  label {
+    cursor: pointer;
+    text-indent: -9999px;
+    width: 38px;
+    height: 20px;
+    background: var(--affine-icon-color);
+    border: 1px solid var(--affine-black-10);
+    display: block;
+    border-radius: 20px;
+    position: relative;
+  }
+
+  label:after {
+    content: '';
+    position: absolute;
+    top: 1px;
+    left: 1px;
+    width: 16px;
+    height: 16px;
+    background: var(--affine-white);
+    border: 1px solid var(--affine-black-10);
+    border-radius: 16px;
+    transition: 0.4s;
+  }
+
+  input:checked + label {
+    background: var(--affine-primary-color);
+  }
+
+  input:checked + label:after {
+    left: calc(100% - 1px);
+    transform: translateX(-100%);
+  }
+
+  label:active:after {
+    width: 24px;
+  }
+`;
+
+export class ToggleSwitch extends LitElement {
```

Comment:
I think maybe the component is used by other components.
so when we use it, Is it supposed to be pass `state` and `onChange` to the switch component?
what do you think?

Reply:
That make sense. I think the 'on' state can be passed in, maybe I need to add onChange.